scripts/githooks/pre-push

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env ruby
+
+require 'net/http'
+require 'uri'
+require 'json'
+
+branch_name = `git rev-parse --abbrev-ref HEAD`
+jira_pattern = /[A-Z]+-\d+/i
+jira_branch_match = jira_pattern.match(branch_name)
+
+if jira_branch_match
+  branch_issue = jira_branch_match[0].upcase
+  jira_creds_dir = File.join(ENV['HOME'], '.jira', 'credentials')
+
+  if File.exist?(jira_creds_dir)
+    creds = Hash[File.read(jira_creds_dir).split("\n").map{|i|i.split('=')}]
+    url = creds['JIRA_BASE_URL']
+    token = creds['JIRA_API_TOKEN']
+    username = creds['JIRA_USER_EMAIL']
+
+    uri = URI.parse("#{url}/rest/api/3/issue/#{branch_issue}")
```

Comment:
Right now we aren't requiring branch name to be correlated to the issue as long as the branches commits have the ID of the issue in them. Would that need to change for this?

Reply:
We'd probably need to move this to pre-commit then, to get the message with the issue id... the pre-push would have to run a git log and loop through all commits

Reply:
I'm hesitant to add anything to the `pre-commit` since we are already seeing some bottle necks for devs there. Using that `git log` strategy in the `pre-push` sounds like a good idea to me. Might have to cop some of the branch naming logic filter matchers from concert / jira validator to get rid of the merges from master

Reply:
@macleanevans I think we should require the branch to have the issue- as we use that to update commit messages? `git log` might not work nicely (tons of output) & we'd have to make an API call to github to get the commits for a PR

Reply:
Good with leaving this. Will have to add docs or a dev section about how to no verify the push and then later rename your branch if you are doing something with a Jira issue yet